### PR TITLE
Backport 2.x: In-tree Python package requirements

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Backport of https://github.com/ARMmbed/mbedtls/pull/5197. Differences with the 3.x version:

* In 2.x, we don't need Python in most builds, and never on Windows. So I kept the changes in `.travis.yml` to a minimum.
* In 2.x, we do not and will not have Python requirements for drivers.

### CODE REVIEW

**Review by mpg — DISMISSED** [review]
LGTM

File: scripts/min_requirements.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                        action='append', dest='pip_general_options',
+                        help="Pass this general option to pip")
+    parser.add_argument('--user',
+                        action='append_const', dest='pip_install_options',
+                        const='--user',
+                        help="Install to the Python user install directory"
+                             " (short for --pip-install-option --user)")
+    parser.add_argument('files', nargs='*', metavar='FILE',
+                        help="Requirement files"
+                             " (default: requirements.txt in the script's directory)")
```
Comment (mprse) [inline scripts/min_requirements.py:122]: Default in code is `ci.requirements.txt`.

**Review by mprse — APPROVED** [review]
Found one minor issues.
Otherwise ok.

**Review by mprse — APPROVED** [review]

**Review by gabor-mezei-arm — APPROVED** [review]
The last commit looks good to me. So taking over the review of @mpg the PR looks good to me.

**Pull request merged**